### PR TITLE
Debugger: Support additional GDB stub packets

### PR DESCRIPTION
**Pull request by Touched**
Add support for GDB memory and register writing commands.

### CODE REVIEW

File: src/debugger/gdb-stub.c [hunk 1]
```diff
@@ -210,6 +211,62 @@ static void _step(struct GDBStub* stub, const char* message) {
...
+
+	if (size > 512) {
+		_error(stub, GDB_BAD_ARGUMENTS);
+		return;
+	}
+
+	struct ARMCore* cpu = stub->d.core->cpu;
+	for (i = 0; i < size; i++) {
+		// Parse escape char
+		uint8_t byte;
```
Comment (endrift) [inline src/debugger/gdb-stub.c]: Please do the assignment here and don't increment the variable in the middle of an expression, for readability.

File: src/debugger/gdb-stub.c [hunk 2]
```diff
@@ -210,6 +211,62 @@ static void _step(struct GDBStub* stub, const char* message) {
...
+		if ((byte = *readAddress++) == 0x7D) {
+			byte = *readAddress++ ^ 0x20;
```
Comment (endrift) [inline src/debugger/gdb-stub.c]: Is this safe? The boundary condition here makes me a bit wary, especially given that we're incrementing with a fixed size that we don't update.

File: src/debugger/gdb-stub.c [hunk 3]
```diff
@@ -243,6 +314,29 @@ static void _readGPRs(struct GDBStub* stub, const char* message) {
...
 }
 
+static void _writeRegister(struct GDBStub* stub, const char* message) {
+	struct ARMCore* cpu = stub->d.core->cpu;
+	const char* readAddress = message;
+
+	unsigned i = 0;
+	uint32_t reg = _readHex(readAddress, &i);
+	readAddress += i + 1;
+	uint32_t value = __builtin_bswap32(_readHex(readAddress, &i));
```
Comment (endrift) [inline src/debugger/gdb-stub.c]: This isn't compatible with MSVC. Please ifdef this and use _byteswap_ulong for MSVC.

**Comment by endrift** [comment]
Looks pretty good so far, but a few minor nitpicks. Please also add this to the Misc section in CHANGES.

File: src/debugger/gdb-stub.c (hunk 2 above)
Comment (Touched) [inline src/debugger/gdb-stub.c]: The size specified as a parameter in the packet is the size in addressable memory units, not the amount of bytes transmitted in the packet. Therefore it's not really possible to check when the packet is over without first reading it. It shouldn't overrun the buffer though, due to the earlier check for `size > 512`. 
Should I make the buffer overrun check more explicit by comparing `size` to `GDB_STUB_MAX_LINE`?

Comment (endrift) [inline src/debugger/gdb-stub.c]: Hmm, I suppose since size <= 512, and the max line size is > 1024, it shouldn't be a problem.

**Comment by endrift** [comment]
Looks good, will merge when I get home.

**Pull request merged**